sw/device/tests/power_virus_systemtest.c

```diff
@@ -469,5 +605,25 @@ bool test_main(void) {
   configure_spi_host();
   spi_device_testutils_configure_passthrough(&spi_device, /*filters=*/0,
                                              /*upload_write_commands=*/false);
+
+  // ***************************************************************************
+  // Kick off test tasks.
+  // ***************************************************************************
+  CHECK(ottf_task_create(crypto_data_load_task, "CryptoDataLoadTask",
+                         kOttfFreeRtosMinStackSize, 1));
+  CHECK(ottf_task_create(comms_data_load_task, "CommsDataLoadTask",
+                         kOttfFreeRtosMinStackSize, 1));
+  CHECK(ottf_task_create(max_power_task, "MaxPowerTask",
+                         kOttfFreeRtosMinStackSize, 1));
+
+  // ***************************************************************************
+  // Yield control flow to the highest priority task in the run queue. Since the
+  // tasks created above all have a higher priority level than the current
+  // "test_main" task, execution will not be returned to the current task until
+  // the above tasks have been deleted.
```

Comment:
This isn't strictly true. If all the higher-priority tasks are blocked (waiting for an event or something), a lower-priority task will run.

Is this test supposed to eventually terminate from the software side?

Reply:
Technically, in an OS theory sense, you are correct hehe, but since we never call a `vTaskDelay()` or equivalent FreeRTOS task blocking [mechanism](https://www.freertos.org/RTOS-task-states.html) anywhere in this test, the lower priority tasks will never let something lower priority run.

Yes this test is supposed to terminate from the SW side.

Theoretically, I could get away without the need for the concurrency framework at all for this test (since I have not hooked up the FreeRTOS task blocking mechanisms to the OTTF yet, and we can mostly kick everything off at once so far it may not be necessary) but I thought it was prudent to have a test that resembles at least parts of a final product as closely as possible, and allows us to have tasks that block on a various things is needed in the future.